dipy/denoise/fast_noise_estimate.pyx

```diff
@@ -0,0 +1,134 @@
+"""
+===========================================
+Estimate Noise Levels for LocalPCA Datasets
+===========================================
+
+"""
+
+import numpy as np
+import nibabel as nib
+import scipy as sp
+from scipy import ndimage
+cimport cython
+cimport numpy as cnp
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+@cython.cdivision(True)
+def fast_noise_estimate(data, gtab):
+    """ Noise estimation for local PCA denoising
+
+    Parameters
+    ----------
+    data: 4D array
+        the input dMR data
+
+    gtab: gradient table object
+        gradient information for the data gives us the
+        bvals and bvecs of diffusion data, which is needed
+        here to select between the noise estimation
+        methods
+
+    Returns
+    -------
+    sigma_corr: 3D array
+        The local noise variance estimate
+
+    References
+    ----------
+    [1]: Manjon JV, Coupe P, Concha L, Buades A, Collins DL
+        "Diffusion Weighted Image Denoising Using Overcomplete
+         Local PCA"
+         
```

Comment:
Oh wait, that is the version where the noise field is estimated from multiple b0s or from the lowest component of the data? In the first case you would need two pca yes (and thus the function could be accessible from outside, invalidating the previous remark.)